doc/releases/changelog-dev.md

```diff
@@ -192,40 +192,6 @@
   [(#6632)](https://github.com/PennyLaneAI/pennylane/pull/6632)
   [(#6653)](https://github.com/PennyLaneAI/pennylane/pull/6653)
 
-<h4>New API for Default mixed</h4>
```

Comment:
FYI this is going to **_fully_** make it into the next release (v0.41) so let's not advertise that it's migrated to the new device API yet.